pages/test_analytics/ruby_collectors.md

```diff
@@ -63,6 +63,14 @@ Failure/Error: allow_any_instance_of(Object).to receive(:sleep)
 
 You can fix them by being more specific in your stubbing by replacing `allow_any_instance_of(Object).to receive(:sleep)` with `allow_any_instance_of(TheClassUnderTest).to receive(:sleep)`.
 
+### Troubleshooting test grouping issues
+
+In RSpec anonymous test cases are supported, tests are automatically named based on the subject and/or inputs to the expectations within the test. However, this can lead to unstable test names across different test runs, incorporating elements such as object IDs, database IDs, timestamps, and more.
+
+As a consequence, each test is assigned a new identity per run within Test Analytics. This poses a challenge for utilising the Test Analytics product effectively, as historical data across tests becomes difficult to track and analyze.
+
+To mitigate this issue and ensure the reliability of Test Analytics, it's advisable to provide explicit and stable descriptions for each test case within your RSpec test suite. By doing so, you can maintain consistency in test identification across multiple runs, enabling better tracking and analysis of test performance over time.
```

Comment:
There a few other solutions available for this that do not require a user to rename all their tests. Should I include any of them here? @jameshill @gchan

Reply:
Was there a work around you found for persona(?) when they encountered this? I think it's valuable to provide that work around here. It'd be good to support customers with massive suites like this.

Reply:
They created their own workaround, but I don't think it would be one we recommend to others as it could affect the TA experience. There was one mentioned here that potentially is a solution https://buildkite-corp.slack.com/archives/C05MSVDKQRG/p1712716872239569?thread_ts=1712702330.510189&cid=C05MSVDKQRG but not sure if that has actually been tested

Reply:
I'm just going to merge this as is. If we come across any examples in the future we can add them in